packages/react-router/lib/dom/ssr/fog-of-war.ts

```diff
@@ -0,0 +1,314 @@
+    async function fetchPatches() {
+      let lazyPaths = getFogOfWarPaths(fogOfWar!, router);
+      if (lazyPaths.length === 0) {
+        return;
+      }
+
+      try {
+        await fetchAndApplyManifestPatches(
+          lazyPaths,
+          fogOfWar!,
+          manifest,
+          routeModules,
+          isSpaMode,
+          router.basename,
+          router.patchRoutes
+        );
+      } catch (e) {
+        console.error("Failed to fetch manifest patches", e);
+      }
+    }
+
+    // Register and fetch patches for all initially-rendered links
+    document.body
+      .querySelectorAll("a[data-discover], form[data-discover]")
+      .forEach((el) => registerElement(el));
+
+    fetchPatches();
+
+    // Setup a MutationObserver to fetch all subsequently rendered links/form
+    let debouncedFetchPatches = debounce(fetchPatches, 100);
+
+    function isElement(node: Node): node is Element {
+      return node.nodeType === Node.ELEMENT_NODE;
+    }
+
+    let observer = new MutationObserver((records) => {
+      let elements = new Set<Element>();
+      records.forEach((r) => {
+        [r.target, ...r.addedNodes].forEach((node) => {
+          if (!isElement(node)) return;
+          if (node.tagName === "A" && node.getAttribute("data-discover")) {
+            elements.add(node);
+          } else if (
+            node.tagName === "FORM" &&
+            node.getAttribute("data-discover")
+          ) {
+            elements.add(node);
+          }
+          if (node.tagName !== "A") {
+            node
+              .querySelectorAll("a[data-discover], form[data-discover]")
+              .forEach((el) => elements.add(el));
+          }
+        });
+      });
+      elements.forEach((el) => registerElement(el));
+      debouncedFetchPatches();
+    });
+
+    observer.observe(document.documentElement, {
+      subtree: true,
+      childList: true,
+      attributes: true,
+      attributeFilter: ["data-discover", "href", "action"],
+    });
+
+    return () => observer.disconnect();
+  }, [isSpaMode, manifest, routeModules, router]);
+}
+
+function getFogOfWarPaths(fogOfWar: FogOfWarInfo, router: Router) {
+  let { knownGoodPaths, known404Paths, nextPaths } = fogOfWar;
+  return Array.from(nextPaths.keys()).filter((path) => {
+    if (knownGoodPaths.has(path)) {
+      nextPaths.delete(path);
+      return false;
+    }
+
+    if (known404Paths.has(path)) {
+      nextPaths.delete(path);
+      return false;
+    }
+
+    return true;
+  });
+}
+
+export async function fetchAndApplyManifestPatches(
+  paths: string[],
+  _fogOfWar: FogOfWarInfo,
+  manifest: AssetsManifest,
+  routeModules: RouteModules,
+  isSpaMode: boolean,
+  basename: string | undefined,
+  patchRoutes: Router["patchRoutes"]
+): Promise<void> {
+  let { nextPaths, knownGoodPaths, known404Paths } = _fogOfWar;
+  let manifestPath = `${basename != null ? basename : "/"}/__manifest`.replace(
```

Comment:
@brophdawg11 why force the path here so the manifest is always at root / path? This breaks apps with the `ssr:false` (in react-router.config.ts) and a `base` path set in vite.config.ts on react-router `7.1.3`. The `manifest.url` has the correct URL but that's not being used.
![image](https://github.com/user-attachments/assets/48044689-6066-4f0d-b1dd-ab781a097ac5)